### ISSUE TITLE
MAX_INITIALIZER_LINES has no effect (Origin: bugzilla #772155)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#6077]
status NEEDINFO severity *normal* in component *general* for *---*
Reported in version *1.8.12* on platform *Other*
Assigned to: Dimitri van Heesch

On 2016-09-29 08:26:51 +0000, max wrote:

    Hello 
    
    I am trying to hide the #define macros in C-Code. As written in
    Documentation, for this MAX_INITIALIZER_LINES should be set to zero. This is
    not working.
    
    BY Setting ENABLE_PREPROCESSING = 0, the define are hidden, but this has
    also lot of side effects
    
    Can you help me?
    
    Best regards
    Max

On 2016-09-29 16:31:51 +0000, albert wrote:

    I think such a question should not be in a bug reporting system but be
    posted in a forum like doxygen-users or stack exchange.
    
    When you think this is a bug please attach a (small) self-contained example
    (source+config file in a tar or zip) that allows us to reproduce the
    problem?

On 2016-09-29 18:45:14 +0000, max wrote:

    Hi I already search in the forums for this problem. No One had an answer,
    that's why I put this in bug reporting system. I will upload soon an example
    with source and config.

**Comment by albert-github** [comment]
The documentation states:
> MAX_INITIALIZER_LINES The MAX_INITIALIZER_LINES tag determines the maximum number of
lines that the initial value of a variable or macro / define can have for it to appear in the documentation.
If the initializer consists of more lines than specified here it will be hidden. Use a value of 0 to hide initializers
completely. The appearance of the value of individual variables and macros / defines can be controlled using
`\showinitializer` or `\hideinitializer` command in the documentation regardless of this setting.

in other words the value / initializer of e.g. the `#define` are hidden but not the `#define` itself.

When you don't want to take the `#define` into account at all you could embed them in a `\cond` / `\endcond` block so that doxygen will skip them completely.
It is also possible not to document them so doxygen won't show them in the documentation either (unless you have `EXTRACT_ALL=YES` set.

As no example is provided and it is unclear what exactly is required the issue is incomplete / usage
